server/controller/src/main/java/ai/starwhale/mlops/domain/runtime/RuntimeService.java

```diff
@@ -462,15 +491,150 @@ public void pull(String projectUrl, String runtimeUrl, String versionUrl, HttpSe
     }
 
     public String query(String projectUrl, String runtimeUrl, String versionUrl) {
-        Long versionId = bundleManager.getBundleVersionId(BundleVersionUrl.create(projectUrl, runtimeUrl, versionUrl));
-        RuntimeVersionEntity runtimeVersionEntity = runtimeVersionMapper.find(versionId);
-        if (null == runtimeVersionEntity) {
+        RuntimeVersionEntity runtimeVersion = (RuntimeVersionEntity) bundleManager.getBundleVersion(
+                BundleVersionUrl.create(projectUrl, runtimeUrl, versionUrl));
+        if (null == runtimeVersion) {
             throw new SwNotFoundException(ResourceType.BUNDLE, "Not found.");
         }
-        return runtimeVersionEntity.getName();
+        return runtimeVersion.getName();
     }
 
     public Boolean recoverRuntime(String projectUrl, String runtimeUrl) {
         throw new UnsupportedOperationException("Please use TrashService.recover() instead.");
     }
+
+    public void buildImage(String projectUrl, String runtimeUrl, String versionUrl) {
+        RuntimeVersionEntity runtimeVersion = (RuntimeVersionEntity) bundleManager.getBundleVersion(
+                BundleVersionUrl.create(projectUrl, runtimeUrl, versionUrl));
+        if (null == runtimeVersion) {
+            throw new SwNotFoundException(ResourceType.BUNDLE_VERSION, "Not found.");
+        }
+        var project = projectService.findProject(projectUrl);
+        var user = userService.currentUserDetail();
+        var builtImage = runtimeVersion.getBuiltImage();
+        if (StringUtils.hasText(builtImage)) {
+            log.debug("runtime:{}-{}'s image:{} has already existed.", runtimeUrl, versionUrl, builtImage);
```

Comment:
Early return is more recommended.  
https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html

Reply:
fixed